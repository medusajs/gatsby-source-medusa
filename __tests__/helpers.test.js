const { processNode } = require("../helpers");

describe("helper functions", () => {
  it("should return return a new node from processNode", () => {
    const fieldName = "product";
    const node = {
      id: "prod_test_1234",
      title: "Test Shirt",
      description: "A test product",
      unit_price: 2500,
    };

    const contentDigest = "digest_string";
    const createContentDigest = jest.fn(() => contentDigest);
    const processNodeResult = processNode(node, fieldName, createContentDigest);
    const expectedNodeContent = JSON.stringify(node);

    expect(createContentDigest).toBeCalled();

    expect(processNodeResult).toMatchSnapshot({
      ...node,
      parent: null,
      children: [],
      internal: {
        type: "MedusaProduct",
        content: expectedNodeContent,
        contentDigest,
      },
    });
  });
});
