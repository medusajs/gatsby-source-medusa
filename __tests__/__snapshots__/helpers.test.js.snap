// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`helper functions should return return a new node from processNode 1`] = `
Object {
  "children": Array [],
  "description": "A test product",
  "id": "prod_test_1234",
  "internal": Object {
    "content": "{\\"id\\":\\"prod_test_1234\\",\\"title\\":\\"Test Shirt\\",\\"description\\":\\"A test product\\",\\"unit_price\\":2500}",
    "contentDigest": "digest_string",
    "type": "MedusaProduct",
  },
  "parent": null,
  "title": "Test Shirt",
  "unit_price": 2500,
}
`;
